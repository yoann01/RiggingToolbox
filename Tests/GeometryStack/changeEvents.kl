
require RiggingToolbox;

operator entry(){

  StartFabricProfiling();

  GeometryStack stack();
  PolygonMeshSphereGenerator sphereGenerator(2.0, 1, true, true);
  PushModifier pushModifier(3.0);
  stack.addGeometryOperator(sphereGenerator);
  stack.addGeometryOperator(pushModifier);

  EvalContext context();
  stack.evaluate(context);
  pushModifier.setPushDist(6.0);
  stack.evaluate(context);

  StopFabricProfiling();

  report( GetEvalPathReport() );
  report( stack.getDesc());
}

