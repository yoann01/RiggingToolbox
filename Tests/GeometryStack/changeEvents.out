function GeometrySet GeometryStack.evaluate!(EvalContext)
--function GeometryCache.update!(io GeometrySet, GeometryOperator)
--function PolygonMeshSphereGenerator.evaluate!(EvalContext, io GeometrySet)
--function GeometryAttributeCache.update!(io GeometrySet, GeometryOperator):["positions"]
----Update:positions
--function PushModifier.evaluate!(EvalContext, io GeometrySet)
function GeometryStack.notify!(Notifier, String, String):PushModifier.changed
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function GeometryAttributeCache.update!(io GeometrySet, GeometryOperator):["positions"]
----Restore:positions
--function PushModifier.evaluate!(EvalContext, io GeometrySet)

GeometryStack {
  geomOperators:[
    {
      type: PolygonMeshSphereGenerator
    }
    {
      type: PushModifier
    }
  ],
  geomSet:   {
    geometries:[
      Sphere:{
      },
    ]
    attributeGenerations:{
      positions:2,
      normals:1,
      uvs0:1,
    },
  }
}
